test/bessel.jl

```diff
@@ -271,4 +271,54 @@ end
             @test f(0,1) ≈ f(0,Complex{Float16}(1))
         end
     end
+
+    @testset "jinc function" begin
+        for T in [Float16, Float32, Float64, Complex{Float16}, Complex{Float32},Complex{Float64}]
+            @test jinc(T(1.8)) ≈ T(-0.116401402395234261196060221416100814277048989838340270527643894)
+            @test jinc(T(0)) ≈ T(1)
+            x = 1.21966989126650445492653884746525517788
+            # check the first root for negative and positive
+            @test jinc(T(x)) + T(1) ≈ T(1)
+            @test jinc(T(-x)) + T(1) ≈ T(1)
+
+
+            # check for symmetry
+            for x = -5.0:1.0:3.0
+                @test jinc(T(10^x)) ≈ jinc(T(- 10^x))
+
+            end
+
+
+            x = T(1e-5)
+            res = T(0.999999999876629944991456407422485505370463)
+            @test jinc(x) ≈ res
```

Comment:
`≈` by default only checks that half of the significant digits match.  I think we want a more stringent test here.

Instead, I would use the `≅` comparison operation defined here https://github.com/JuliaMath/SpecialFunctions.jl/blob/9116108ba52d103a05d47d9af1ca6b959a536c30/test/runtests.jl#L13

Reply:
Didn't know about that one, but I'll try to fix it.

Reply:
We'll need to update `≅` so that it uses a different tolerance for single precision.  Something like:
```jl
checktol(err::Float32) = err ≤ 1e-6
checktol(err::Float64) = err ≤ 1e-13
≅(a,b) = checktol(relerrc(a,b))
```